ros_ws/src/autonomous/neuralnetwork/include/ai_vector_math.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <algorithm>
+#include <cstdlib>
+#include <vector>
+
+inline void add(std::vector<fann_type>& a, std::vector<fann_type>& b)
+{
```

Comment:
You could add a assertion here to make sure that `b` has at least the same size as `a` e.g.
```C++
ROS_ASSERT_MSG(b.size() >= a.size(), "...");
```